scopes/lanes/ui/compare/lane-compare/lane-compare.tsx

```diff
@@ -0,0 +1,184 @@
+    [base, compare]
+  );
+  const commonComponents = useMemo(
+    () =>
+      compare.components
+        .filter((componentId) => {
+          const compIdStr = componentId.toStringWithoutVersion();
+          const baseCompId = baseMap.get(compIdStr);
+          return baseCompId && !baseCompId.isEqual(componentId);
+        })
+        .map((cc) => [
+          baseMap.get(cc.toStringWithoutVersion()) as ComponentID,
+          compareMap.get(cc.toStringWithoutVersion()) as ComponentID,
+        ]),
+    [base.components, compare.components]
+  );
+
+  const allComponents = useMemo(() => [...commonComponents, ...uniqueCompare], [base.components, compare.components]);
+
+  const defaultState = useCallback(([_base, _compare]: [ComponentID | undefined, ComponentID | undefined]) => {
+    const _tabs = extractLazyLoadedData(tabs)?.sort(sortTabs);
+
+    const value: ComponentCompareState = {
+      tabs: {
+        controlled: true,
+        id: _tabs && _tabs[0].id,
+        element: _tabs && _tabs[0].element,
+      },
+      code: {
+        controlled: true,
+      },
+      aspects: {
+        controlled: true,
+      },
+      versionPicker: {
+        element: (
+          <div className={styles.versionPicker}>
+            {`${(_base && 'Comparing Component') || 'Component'} ${_compare?.toStringWithoutVersion()}`}
+            <div>{`${_compare?.version} ${(_base && ` with ${_base.version}`) || ''}`}</div>
+          </div>
+        ),
+      },
+    };
+    return value;
+  }, []);
+
+  const [state, setState] = useState<LaneCompareState>(
+    new Map(
+      allComponents.map(([_base, _compare]) => {
+        const key = computeStateKey(_base, _compare);
+        const value = defaultState([_base, _compare]);
+        return [key, value];
+      })
+    )
+  );
+
+  const [closeDrawerList, onToggleDrawer] = useState<string[]>([]);
+
+  const handleDrawerToggle = (id: string) => {
+    const isDrawerOpen = !closeDrawerList.includes(id);
+    if (isDrawerOpen) {
+      onToggleDrawer((list) => list.concat(id));
+      return;
+    }
+    onToggleDrawer((list) => list.filter((drawer) => drawer !== id));
+  };
+
+  const hooks = useCallback((_base?: ComponentID, _compare?: ComponentID) => {
+    const key = computeStateKey(_base, _compare);
+    const _tabs = extractLazyLoadedData(tabs);
+
+    const onClicked = (prop: ComponentCompareStateKey) => (id) =>
+      setState((value) => {
+        let existingState = value.get(key);
+        const propState = existingState?.[prop];
+        if (propState) {
+          propState.id = id;
+          propState.element = _tabs?.find((_tab) => _tab.id === id)?.element;
+        } else {
+          existingState = defaultState([_base, _compare]);
+        }
+        return new Map(value);
+      });
+
+    const _hooks: ComponentCompareHooks = {
+      code: {
+        onClick: onClicked('code'),
+        useUpdatedUrlFromQuery: () => useUpdatedUrlFromQuery({}),
```

Comment:
are you sure we need this? cant just handle it internally? would make a better api